File: example-projects/scala/src/test/scala/com/tngtech/jgiven/scala/example/ThenStage.scala
package com.tngtech.jgiven.scala.example

import com.tngtech.jgiven._
import com.tngtech.jgiven.annotation._
import org.scalatest.Assertions._

class ThenStage extends Stage[ThenStage] {

  @ExpectedScenarioState
  var result = 0

  def the_result_should_be(expectedResult: Int): ThenStage = {
    assert(result == expectedResult)
    this
  }

}

File: example-projects/testng/src/test/java/com/tngtech/jgiven/example/projects/testng/ThenStage.java
package com.tngtech.jgiven.example.projects.testng;

import com.tngtech.jgiven.Stage;
import com.tngtech.jgiven.annotation.ScenarioState;
import org.testng.asserts.Assertion;

class ThenStage extends Stage<ThenStage> {
    @ScenarioState(required = true)
    String result;

    public void the_result_is(String expectedResult) {
        new Assertion().assertEquals(expectedResult, result);
    }
}
